src/containers/Missions/missionSlice.js

```diff
@@ -10,21 +12,46 @@ export const missionSlice = createSlice({
   name: 'missions',
   initialState,
   reducers: {
-    joinMission: (state, action) => ({
-      ...state,
-      joinedMissions: [...state.joinedMissions, action.payload],
-      totalJoinedMissions: state.totalJoinedMissions + 1,
-    }),
-    leaveMission: (state, action) => ({
+    joinMission: (state, action) => {
+      // console.log(action.payload);
+      const newState = [...state.missions];
+      const updatedMissions = newState.map((mission) => {
+        if (mission.id !== action.payload) {
+          return mission;
+        }
+        return { ...mission, joined: !mission.joined };
```

Comment:
👍  I would suggest to move the store to separate folder. I will merge this pull request and for that I will work on that